### PR TITLE
Change CustomIdentity to use underscores during encoding

### PR DESCRIPTION
**Pull request by nporter-adbe**
https://github.com/adobe/aepsdk-core-ios/issues/433

### CODE REVIEW

**Comment by emdobrin** [comment]
We should have tests validating the format.
Secondly, what happens with previously migrated ids, will those be lost in 3.0.0?

**Comment by nporter-adbe** [comment]
@emdobrin that's a good point about the migration aspect, I'm not sure if changing the coding keys will have an impact. I can run a quick test after PTO.

**Comment by nporter-adbe** [comment]
@emdobrin So I don't believe we need to worry about migrating. The initializer for the CustomIdentity wasn't accessible publicly, so I don't believe it would have been possible to go through this workflow.

**Comment by nporter-adbe** [comment]
Will be adding tests shortly.

**Comment by nporter-adbe** [comment]
@emdobrin, I did confirm that changing the coding keys will break loading it from persistence. Given that I don't believe there was any way to set `CustomIdentity` in the SDK, given the init level was internal, I'm not sure we need to worry about that here.

File: AEPIdentity/Sources/CustomIdentity.swift [hunk 1]
```diff
@@ -24,19 +24,46 @@ public class CustomIdentity: Identifiable, Codable {
...
+        do {
+            // Attempt to decode using the correct keys
+            let values = try decoder.container(keyedBy: CodingKeys.self)
+            origin = try? values.decode(String.self, forKey: .origin)
+            type = try? values.decode(String.self, forKey: .type)
+            identifier = try? values.decode(String.self, forKey: .identifier)
+            authenticationState = try values.decode(MobileVisitorAuthenticationState.self, forKey: .authenticationState)
+        } catch _ {
+            // Decoding failed using correct keys, attempt to decode using migration keys
+            let values = try decoder.container(keyedBy: MigrationCodingKeys.self)
```
Comment (nporter-adbe) [inline AEPIdentity/Sources/CustomIdentity.swift:60]: If decoding fails we attempt to use the old "Migration" keys. @emdobrin

Comment (emdobrin) [reply AEPIdentity/Sources/CustomIdentity.swift:60]: I think we should either include this in the migration class or expect a new sync call from the developer for the lost ids.  @shalehaha what do you think? this manifests if migrating from v4/v5->3.0.0.

File: AEPIdentity/Sources/CustomIdentity.swift [hunk 2]
```diff
@@ -24,18 +24,18 @@ public class CustomIdentity: Identifiable, Codable {
     ///   - type: Type of the identifier
     ///   - identifier: The identifier
     ///   - authenticationState: Authentication state for the identifier
-    init(origin: String?, type: String?, identifier: String?, authenticationState: MobileVisitorAuthenticationState) {
+    public init(origin: String?, type: String?, identifier: String?, authenticationState: MobileVisitorAuthenticationState) {
```
Comment (emdobrin) [inline AEPIdentity/Sources/CustomIdentity.swift:27]: Why do we need this class to be public? Looking at the Identity public APIs it looks like the sync apis accept free form dictionaries, while the getter returns Identifiable.

Comment (nporter-adbe) [reply AEPIdentity/Sources/CustomIdentity.swift]: fixed.

Comment (emdobrin) [reply AEPIdentity/Sources/CustomIdentity.swift]: can we remove the modifier from the class too? cc @shalehaha

Comment (nporter-adbe) [reply AEPIdentity/Sources/CustomIdentity.swift]: removed that too. the original intention was for this class to be public, but that has been replaced with the `Identifiable` protocol.

**Review by shalehaha — APPROVED** [review]

File: AEPIdentity/Sources/CustomIdentity.swift (hunk 2 above)
Comment (shalehaha) [reply AEPIdentity/Sources/CustomIdentity.swift]: yep, should change it to private

**Review by emdobrin — APPROVED** [review]

**Pull request merged**